rust/connlib/tunnel/src/device_channel.rs

```diff
@@ -64,7 +64,23 @@ impl Device {
         }
     }
 
-    #[cfg(any(target_os = "android", target_os = "linux"))]
+    #[cfg(target_os = "android")]
```

Comment:
Out of curiosity, why define a new function and not just conditionally execute only the `on_set_...` callback for android?

P.s. see @conectado 's comment -- https://firezonehq.slack.com/archives/C04HRQTFY0Z/p1716397330795009?thread_ts=1716366377.144259&cid=C04HRQTFY0Z

Reply:
I'm not sure how to do `cfg` blocks inside a function without defining another function. I saw conectado's comment but didn't look into it yet. It's true this duplicates a few lines, but it should revert to known behavior on Android, I thought we might be in a hurry to fix it.